### PR TITLE
Pipedream SDK

### PR DESCRIPTION
**Pull request by dylburger**
## Summary by CodeRabbit

- **New Features**
  - Added a React component for user authentication and GitHub interaction in a Next.js application.
  - Introduced methods for connecting applications and managing user sessions.
  - Enhanced user experience with responsive rendering based on authentication state.
  - Launched a client module for token creation and account management with structured API interactions.
  - Introduced a new example `.env` file to aid in environment variable configuration.
  - Added a new README file to guide developers in using the Pipedream SDK.
  - Established a comprehensive TypeScript configuration for both browser and Node.js environments.

- **Bug Fixes**
  - Improved error handling by validating necessary environment variables for authentication.

- **Documentation**
  - Included an example `.env` file to assist in environment variable configuration.
  - Updated the README with installation instructions and usage examples for the Pipedream SDK.

### CODE REVIEW

File: packages/sdk/examples/next-app/app/CodePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      <div className="absolute top-0 left-0 right-0 p-2 flex items-center bg-gray-900 rounded-md">
+        <div className="flex space-x-2 ml-2 my-1">
+          <span className="block w-3 h-3 bg-red-500 rounded-full"></span>
+          <span className="block w-3 h-3 bg-yellow-400 rounded-full"></span>
+          <span className="block w-3 h-3 bg-green-500 rounded-full"></span>
+        </div>
+      </div>
+      <div className="pt-6 mt-2 overflow-x-auto">
+        <pre>
+          <code dangerouslySetInnerHTML={{ __html: highlightedCode }} />
```
[the remark this answers is not in the document]
Comment (dylburger) [reply packages/sdk/examples/next-app/app/CodePanel.tsx:42]: awesome! done.

File: packages/sdk/examples/next-app/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    if (!externalUserId || !token) return
+    setOauthAppId(app)
+    pd.startConnect({
+      app,
+      token,
+      onSuccess: ({ id: authProvisionId }: any) => {
+        setAuthProvisionId(authProvisionId as string)
+      }
+    })
+  }
```
[the remark this answers is not in the document]
Comment (dylburger) [reply packages/sdk/examples/next-app/app/page.tsx:53]: fixed

File: packages/sdk/dist/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+};
+export function createClient(opts) {
+    return new ServerClient(opts);
+}
+class ServerClient {
+    constructor(opts) {
+        this.environment = opts.environment;
+        this.secretKey = opts.secretKey;
+        this.publicKey = opts.publicKey;
+        const { apiHost = "pipedream.com" } = opts;
```
Comment (celador) [inline packages/sdk/dist/index.js:18]: Needs to be changed to api.pipedream.com

**Review by celador — CHANGES_REQUESTED** [review]

File: packages/sdk/dist/index.js (hunk 3 above)
Comment (dylburger) [reply packages/sdk/dist/index.js:18]: done in new branch

File: components/_8x8_connect/_8x8_connect.app.mjs [hunk 4]
```diff
@@ -8,4 +8,4 @@ export default {
       console.log(Object.keys(this.$auth));
     },
   },
-};
```
Comment (dylburger) [inline components/_8x8_connect/_8x8_connect.app.mjs:11]: Ignore all of these component changes. We should fix the underlying ESLint config so this doesn't happen, not urgent for this PR

**Pull request merged**